### PR TITLE
Awais786/bom 1240

### PR DESCRIPTION
**Pull request by awais786**
Make openedxstats OEP-18 compliant
Update the django to 1.11 also.
https://openedx.atlassian.net/browse/BOM-1240

### CODE REVIEW

**Comment by awais786** [comment]
```
ERROR: test_todays_logs (openedxstats.apps.sites.tests.ReferrerLogTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arbisoft/Documents/devstack/openedxstats/openedxstats/apps/sites/tests.py", line 547, in setUp
    self.conn = boto.connect_s3()
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/__init__.py", line 141, in connect_s3
    return S3Connection(aws_access_key_id, aws_secret_access_key, **kwargs)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/s3/connection.py", line 191, in __init__
    validate_certs=validate_certs, profile_name=profile_name)
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/connection.py", line 569, in __init__
    host, config, self.provider, self._required_auth_capability())
  File "/Users/arbisoft/.virtualenvs/openedxstats_3/lib/python3.5/site-packages/boto/auth.py", line 991, in get_auth_handler
    'Check your credentials' % (len(names), str(names)))
boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
```

File: requirements/base.in [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+colorama
+dj-database-url
+Django
+django-bootstrap3
+django-datetime-widget
+django-dotenv
+django-filter
+django-model-utils
+djangorestframework
+gunicorn
```
Comment (Ayub-Khan) [inline requirements/base.in]: should go to production.in

File: requirements/base.in [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+lazy-object-proxy
+ordereddict
+psycopg2
+Pygments
+pylint
```
Comment (Ayub-Khan) [inline requirements/base.in]: should go to test.in

File: requirements/base.in [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+django-bootstrap3
+django-datetime-widget
+django-dotenv
+django-filter
+django-model-utils
+djangorestframework
+gunicorn
+lazy-object-proxy
+ordereddict
+psycopg2
```
Comment (Ayub-Khan) [inline requirements/base.in]: should go to production.

File: requirements/base.in [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Pygments
+pylint
+python-dateutil
+python-dotenv
+pytz
+requests
+slacker
+urwid
+whitenoise
```
Comment (Ayub-Khan) [inline requirements/base.in]: should go to production.in

File: openedxstats/apps/sites/tests.py [hunk 5]
```diff
@@ -335,6 +336,16 @@ def test_get_blank_geozone_form(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(0, GeoZone.objects.count())
 
+    def test_page_is_accessible_url(self):
+        """verify all page urls are working fine."""
+        for urls in [
+            'sites:sites_list', 'sites:sites_list_json', 'sites:sites_map',
+            'sites:hawthorn_map', 'sites:add_site',
+            'sites:add_language', 'sites:add_geozone', 'login'
```
Comment (jmbowman) [inline openedxstats/apps/sites/tests.py]: Thank you for adding this!

**Pull request merged**